src/fingerprint_bond.cpp

```diff
@@ -0,0 +1,917 @@
+	if (strcmp(constant,"re")==0){
+		re = strtod(words[0],NULL);
+	}
+	else if (strcmp(constant,"rc")==0){
+		rc = strtod(words[0],NULL);
+	}
+	else if (strcmp(constant,"alphak")==0){
+		delete [] alpha_k;
+		alpha_k = new double [nwords];
+		for (l=0;l<nwords;l++){
+			alpha_k[l]=strtod(words[l],NULL);
+		}
+	}
+	else if (strcmp(constant,"dr")==0){
+		dr = strtod(words[0],NULL);
+	}
+	else if (strcmp(constant,"k")==0){
+		k = strtol(words[0],NULL,10);
+	}
+	else if (strcmp(constant,"m")==0){
+		m = strtol(words[0],NULL,10);
+	}
+	else pair->errorf("Undefined value for bond power");
+	delete [] words;
+	if (re!=0.0 && rc!=0.0 && alpha_k[0]!=-1 && dr!=0.0 && m!=0 && k!=0)return true;
+	return false;
+}
+
+void Fingerprint_bond::write_values(FILE *fid){
+	int i;
+	fprintf(fid,"fingerprintconstants:");
+	fprintf(fid,"%s",pair->elementsp[atomtypes[0]]);
+	for (i=1;i<n_body_type;i++){
+		fprintf(fid,"_%s",pair->elementsp[atomtypes[i]]);
+	}
+	fprintf(fid,":%s_%d:re:\n",style,id);
+	fprintf(fid,"%f\n",re);
+	fprintf(fid,"fingerprintconstants:");
+	fprintf(fid,"%s",pair->elementsp[atomtypes[0]]);
+	for (i=1;i<n_body_type;i++){
+		fprintf(fid,"_%s",pair->elementsp[atomtypes[i]]);
+	}
+	fprintf(fid,":%s_%d:rc:\n",style,id);
+	fprintf(fid,"%f\n",rc);
+	fprintf(fid,"fingerprintconstants:");
+	fprintf(fid,"%s",pair->elementsp[atomtypes[0]]);
+	for (i=1;i<n_body_type;i++){
+		fprintf(fid,"_%s",pair->elementsp[atomtypes[i]]);
+	}
+	fprintf(fid,":%s_%d:alphak:\n",style,id);
+	for (i=0;i<k;i++){
+		fprintf(fid,"%f ",alpha_k[i]);
+	}
+	fprintf(fid,"\n");
+	fprintf(fid,"fingerprintconstants:");
+	fprintf(fid,"%s",pair->elementsp[atomtypes[0]]);
+	for (i=1;i<n_body_type;i++){
+		fprintf(fid,"_%s",pair->elementsp[atomtypes[i]]);
+	}
+	fprintf(fid,":%s_%d:dr:\n",style,id);
+	fprintf(fid,"%f\n",dr);
+	fprintf(fid,"fingerprintconstants:");
+	fprintf(fid,"%s",pair->elementsp[atomtypes[0]]);
+	for (i=1;i<n_body_type;i++){
+		fprintf(fid,"_%s",pair->elementsp[atomtypes[i]]);
+	}
+	fprintf(fid,":%s_%d:k:\n",style,id);
+	fprintf(fid,"%d\n",k);
+	fprintf(fid,"fingerprintconstants:");
+	fprintf(fid,"%s",pair->elementsp[atomtypes[0]]);
+	for (i=1;i<n_body_type;i++){
+		fprintf(fid,"_%s",pair->elementsp[atomtypes[i]]);
+	}
+	fprintf(fid,":%s_%d:m:\n",style,id);
+	fprintf(fid,"%d\n",m);
+}
+
+void Fingerprint_bond::init(int *i,int id){
+	for (int j=0;j<n_body_type;j++){atomtypes[j] = i[j];}
+	re = 0;
+	rc = 0;
+	m = 0;
+	this->k = 0;
+	alpha_k = new double [1];
+	alpha_k[0]=-1;
+	empty = false;
+	this->id = id;
+}
+
+//number of neurons defined by this fingerprint
+int Fingerprint_bond::get_length(){
+	return m*k;
+}
+
+void Fingerprint_bond::allocate(){
+	generate_exp_cut_table();
+	generate_coefficients();
+	generate_rinvssqrttable();
+}
+
+//Generate table of complex functions for quick reference during compute. Used by do3bodyfeatureset_singleneighborloop and do3bodyfeatureset_doubleneighborloop.
+void Fingerprint_bond::generate_exp_cut_table(){
+  int m,n;
+	double r1;
```

Comment:
there is inconsistent indentation here. as a reminder, in LAMMPS indentations are 2 spaces per level, no TABs, and no whitespace at the end of a line.